// @aws-sdk/client-ecs/dist-es/commands/ListTaskDefinitionFamiliesCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTaskDefinitionFamiliesRequestFilterSensitiveLog, ListTaskDefinitionFamiliesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListTaskDefinitionFamiliesCommand, serializeAws_json1_1ListTaskDefinitionFamiliesCommand, } from "../protocols/Aws_json1_1";
export class ListTaskDefinitionFamiliesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListTaskDefinitionFamiliesCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ECSClient";
        const commandName = "ListTaskDefinitionFamiliesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListTaskDefinitionFamiliesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListTaskDefinitionFamiliesResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1ListTaskDefinitionFamiliesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1ListTaskDefinitionFamiliesCommand(output, context);
    }
}
